### PR TITLE
html code : form

### LINKED ISSUE #1: create form fields
# setup and dev server
1. run in shell/terminal
```
git clone https://github.com/kevinnls/eflufeepaymentdetails.git -b form efpd
cd efpd
npm install
npm run dev
```
2. go to http://localhost:5173/

# file to work on
```
src/routes/form/+page.svelte
```

# things to note
1. use the appropriate [`<input>` type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types) or `<select>` as necessary
2. every `<input>` should have an `id=` attribute
3. every input field should have [`<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) with the right [`for=` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attributes)
5. it should look ugly. don't style it. refer <https://motherfuckingwebsite.com/>

# form reference

these are the details to be collected from the user. need html form with the right fields

image

[[source]](https://efluniversity.ac.in/Payment%20of%20semester%20fees%20for%20the%20UG%20Programmes%20Aug-2023-New%205th%20Sem.pdf)

### PR DESCRIPTION
**Pull request by Mckenz31**
closes #1

### CODE REVIEW

File: src/routes/form/+page.svelte [hunk 1]
```diff
@@ -23,4 +25,61 @@
...
+	<input type="text" id="acc-holder" name="acc-holder">
+	<label for="acc-no">Account No</label>
+	<input type="number" id="acc-no" name="acc-no">
+	<label for="bank">Name of the bank</label>
+	<input type="text" id="bank" name="bank">
+	<label for="branch">Branch</label>
+	<input type="text" id="branch" name="branch">
+
+	<label for="ifsc">IFSC code</label>
+	<input type="number" id="ifsc" name="ifsc">
```
Comment (kevinnlsamuel) [inline src/routes/form/+page.svelte]: ifsc can't be number :joy:  first four characters are bank code and they are letters

File: src/routes/form/+page.svelte [hunk 2]
```diff
@@ -23,4 +25,61 @@
...
+	<input type="number" id="reference-no" name="reference-no">
+	<label for="utr">UTR Number</label>
+	<input type="number" id="utr" name="utr">
+	<label for="inb">INB Reference Number (for transfer from SBI to SBI)</label>
+	<input type="number" id="inb" name="inb">
+
+	<strong>Details of the fees payment</strong>
+	<br>
+
+	<ol type = "a">
```
Comment (kevinnlsamuel) [inline src/routes/form/+page.svelte]: type a?

File: src/routes/form/+page.svelte [hunk 3]
```diff
@@ -23,4 +25,61 @@
 		<option value="9">Sem IX</option>
 		<option value="10">Sem X</option>
 	</select>
+	<label for="role">Roll No</label>
+	<input type = "text" id = "role" name = "role">
```
Comment (kevinnlsamuel) [inline src/routes/form/+page.svelte]: *`roll`

File: src/routes/form/+page.svelte [hunk 4]
```diff
@@ -23,4 +25,61 @@
...
+	<label for="category">Category</label>
+	<select id="category">
+		<option value="1">General</option>
+		<option value="2">EWS</option>
+		<option value="3">OBC</option>
+		<option value="4">WDP</option>
+		<option value="5">SC</option>
+		<option value="6">ST</option>
+		<option value="7">VHC</option>
+		<option value="8">PHC</option>
```
Comment (kevinnlsamuel) [inline src/routes/form/+page.svelte]: comprehensible `value`s would be better here: `gen`, `ews`, etc

**Review by kevinnlsamuel — CHANGES_REQUESTED** [review]
okay i forgot how to do this too. but in some of the places, you've used `type=number` we have a problem because it adds the arrows and users tend to accidentally increment/decrement values (myself included)

from https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#accessibility
>The implicit [role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) for the <input type="number"> element is [spinbutton](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/spinbutton_role). If spinbutton is not an important feature for your form control, consider not using type="number". Instead, use [inputmode="numeric"](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode) along with a [pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern) attribute that limits the characters to numbers and associated characters. With <input type="number">, there is a risk of users accidentally incrementing a number when they're trying to do something else. Additionally, if users try to enter something that's not a number, there's no explicit feedback about what they're doing wrong.

so i think `<input type="text" inputmode="numeric">` is what we need

you can see this especially when testing on mobile, the software keyboard would be different

and i think we may have to keep bank reference "numbers" as the default `<input type="text">` they may include characters

File: src/routes/form/+page.svelte (hunk 2 above)
Comment (Mckenz31) [reply src/routes/form/+page.svelte]: From what I recollect, type = "a" orders the list indexes as a. b. c. and so on

File: src/routes/form/+page.svelte (hunk 1 above)
Comment (Mckenz31) [reply src/routes/form/+page.svelte]: Did not know that. Will change

File: src/routes/form/+page.svelte (hunk 4 above)
Comment (Mckenz31) [reply src/routes/form/+page.svelte]: Didn't get you

File: src/routes/form/+page.svelte (hunk 2 above)
Comment (kevinnlsamuel) [reply src/routes/form/+page.svelte]: hahaha. no. that's a CSS prop. `style="list-style-type: lower-roman-latin`. or setting it in CSS. in tailwind it's `class="...list-[lower-latin]...`

but eitherway, not required. just remove it

File: src/routes/form/+page.svelte (hunk 4 above)
Comment (kevinnlsamuel) [reply src/routes/form/+page.svelte]:
```
<option value="gen">General</option>
<option value="ews">EWS</option>
...
```

File: src/routes/form/+page.svelte [hunk 5]
```diff
@@ -67,18 +67,18 @@
...
 	</select>
 	<label for="transaction-no">Transaction Number</label>
-	<input type="number" id="transaction-no" name="transaction-no">
+	<input type="text" inputmode = "numeric" id="transaction-no" name="transaction-no">
 	<label for="reference-no">Reference Number</label>
-	<input type="number" id="reference-no" name="reference-no">
+	<input type="text" id="reference-no" name="reference-no">
 	<label for="utr">UTR Number</label>
-	<input type="number" id="utr" name="utr">
+	<input type="text" inputmode = "numeric" id="utr" name="utr">
```
Comment (kevinnlsamuel) [inline src/routes/form/+page.svelte]: not numeric

File: src/routes/form/+page.svelte [hunk 6]
```diff
@@ -67,18 +67,18 @@
 		<option value="4">Credit card</option>
 	</select>
 	<label for="transaction-no">Transaction Number</label>
-	<input type="number" id="transaction-no" name="transaction-no">
+	<input type="text" inputmode = "numeric" id="transaction-no" name="transaction-no">
```
Comment (kevinnlsamuel) [inline src/routes/form/+page.svelte]: not numeric

**Review by kevinnlsamuel — COMMENTED** [review]
this and the values for category

**Review by kevinnlsamuel — APPROVED** [review]

**Comment by kevinnlsamuel** [comment]
all done! @Mckenz31 can you do a squash merge please?

**Comment by kevinnls** [comment]
closes #1

**Pull request merged**